e2e: updating PR pipeline (#19478)
status-im/status-mobile: doc/pipeline_process.md

### What if the work is still in progress?

- If PR work is not finished yet, please mark it as a draft or add [WIP] to the title and keep it in the `CONTRIBUTOR` column until it's ready to be reviewed/tested.
- Alternatively, you can open a `draft`.

### When is a PR considered ready to move forward?

Ready for testing, a PR should meet the following criteria:

1. Reviewed and has at least 1 approval.
2. Rebased to the `develop` branch (both `status-mobile` and `status-go` if needed, depending on what part has changes).

>    - **NOTE:** QAs can use the `Update with rebase` button from GitHub in case if:

Reviewer: Minor grammatical thing

```suggestion
   - **NOTE:** QAs can use the `Update with rebase` button from GitHub if:
```

or 

```suggestion
   - **NOTE:** QAs can use the `Update with rebase` button from GitHub in case of:
```

        - The PR has no `status-go` changes.
        - The PR has no conflict with the develop branch.
3. All possible conflicts have been resolved.
4. Has the label: `request-manual-qa` or `skip-manual-qa`.
5. PRs **MUST** identify what area is affected and should have a description.

**NOTE:** Make sure that QAs are OK with that

### Adding `skip-manual-qa`

> - Please ask another team member before adding the `skip-manual-qa` label (or here in chat) so that there's a second opinion.

Reviewer 2: > (or here in chat)

looks like it should be `(or on Discord/Status community/DMs)` or smth like that instead

> - The PR should have a proper reasoning why manual QA is skipped.

Reviewer: ```suggestion
- The PR MUST have a proper reasoning why manual QA is skipped.
```

> - The PR should include the steps of testing that has been done by the developer prior to moving it forward.

Reviewer: I don't know if this one needs to be a MUST, but I've suggested it

```suggestion
- The PR MUST include the steps of testing that has been done by the developer prior to moving it forward.
```

**From the perspective of a developer it means that once work on PR is finished:**

1. It should be rebased to the latest `develop`. If there are conflicts - they should be resolved if possible.
2. If the PR was in the `Contributor` column - it should be moved to `Review` column.
3. Wait for the review.
4. Make sure that after review and before requesting manual QA your PR is rebased to current develop.

> 6. **MUST:** Move PR to the E2E column when it is ready for testing (**mandatory for all PRs**). That will also trigger e2e tests run. QAs are monitoring PRs from E2E column and take it into test. 

Reviewer: This makes more grammatical sense to me, but I also may be misunderstanding.

If the full sentence is

> From the perspective of a developer it means that once work on PR is finished:

Then the below sentence makes more sense to me.

```suggestion
6. The PR **MUST** be moved to the E2E column when it is ready for testing (**mandatory for all PRs**). That will also trigger e2e tests run. QAs are monitoring PRs from E2E column and take it into test. 
```

After that - PR will be taken into manual testing by the QA team.

### E2E tests and analyzing the results

This step cannot be skipped. So, at least one comment from the `status-im-auto` bot with results is a prerequisite for moving forward.
Information on how to analyze tests can be found [here](https://github.com/status-im/status-mobile/blob/develop/doc/how-to-launch-e2e.md).
Tests might be flaky, as they depend on infrastructure - SauceLabs and Waku.

> If there are `Failed tests` and you are not sure about the reason, you can always ping the mobile QAs for help (preferably in PRs).

Reviewer 2: I think you can also mention here that devs should use our group tag `@status-im/mobile-qa` in this case

**For now, it is the only reliable source of regression in the app because, due to time constraints, it is not possible to test thoroughly each PR.
Please, respect this rule.**

## Testing PR

### Manual testing
#### Prerequisites for manual testing

> 1. PR has a description and the area affected.
> 2. PR has automation results from the previous step.
> 3. PR includes a demo, test steps, and possible regression
> 4. PR has out of scope (if appropriate)

Reviewer: ```suggestion
1. PR has a description of the change and details the area of the application affected by the change.
2. PR has automation results from the previous E2E test run step.
3. PR includes a demo, test steps, and any possible regression(s)
4. If appropriate, PR details what elements of a feature are out of scope.
```

---        

#### Why my PR is in `Contributor` column?

> PR can be moved to this column by the ```status-github-bot``` or by QA engineer with label `Tested-issues` or if one of the requirement for manual QA was not met.

Reviewer: ```suggestion
PR can be moved to this column by the ```status-github-bot``` or by QA engineer with label `Tested-issues` or if one of the requirements for manual QA was not met.
```
